toqito/channel_metrics/fidelity_of_separability.py

```diff
@@ -0,0 +1,198 @@
+        psi_dims: the dimensions of System A, B & R in
+            the input state density matrix. It is assumed that the first
+            quantity in this list is the dimension of System B.
+        k: value for k-extendibility
+    Raises:
+        AssertionError:
+            * If the provided dimensions are not for a tripartite density \n
+            matrix
+        TypeError:
+            * If the matrix is not a density matrix (square matrix that is \n
+            PSD with trace 1).
+        TypeError:
+            * If the input state is a mixed state.
+    Returns:
+        Optimized value of the SDP when maximized over a set of linear
+        operators subject to some constraints.
+    """
+    if not is_density(psi):
+        raise ValueError("Provided input state is not a density matrix.")
+    if not len(psi_dims) == 3:
+        raise AssertionError("For Channel SDP: require tripartite state dims.")
+    if not is_pure(psi):
+        raise ValueError("This function only works for pure states.")
+
+    # We first permure psi_{BAR} to psi_{RAB} to simplify the code.
+    psi = permute_systems(psi, [3, 2, 1], psi_dims)
+    dim_b, dim_a, dim_r = psi_dims
+    psi_dims = [dim_r, dim_a, dim_b]
+
+    # List of dimensions [R, A, B, A'] needed to define optimization function.
+    dim_list = [dim_r, dim_a, dim_b, dim_a]
+
+    # Dimension of the Choi matrix of the extended channel.
+    choi_dims = [dim_r] + [dim_a] * k
+
+    # List of extenstion systems and dimension of the Choi matrix.
+    sys_ext = list(range(2, 2 + k - 1))
+    dim_choi = dim_r * (dim_a**k)
+
+    # Projection onto symmetric subspace on AA'.
+    pi_sym = symmetric_projection(dim_a, 2)
+
+    choi = picos.HermitianVariable("S", (dim_choi, dim_choi))
+    choi_partial = picos.partial_trace(choi, sys_ext, choi_dims)
+    sym_choi = symmetric_projection(dim_a, k)
+    problem = picos.Problem(verbosity=2)
+
+    problem.set_objective(
+        "max",
+        np.real(picos.trace(
+            pi_sym *
+            picos.partial_trace(
+                (picos.partial_transpose(
+                    psi, [0], psi_dims) @ picos.I(dim_a)) *
+                permute_systems(
+                    choi_partial @ picos.I(dim_b*dim_a), [
+                        1, 4, 3, 2], dim_list),
+                [0, 2], dim_list
+            )
+        ))
+    )
+
+    problem.add_constraint(
+        picos.partial_trace(choi, list(range(
+            1, k+1)), choi_dims) == picos.I(dim_r)
+    )
+    problem.add_constraint(choi >> 0)
+
+    # k-extendablility of Choi state
+    problem.add_constraint(
+        (picos.I(dim_r) @ sym_choi) * choi * (picos.I(dim_r)@sym_choi) == choi
+    )
+
+    # PPT condition on Choi state
+    sys = []
+    for i in range(1, 1+k):
+        sys = sys + [i]
+        problem.add_constraint(
+            picos.partial_transpose(choi, sys, choi_dims) >> 0)
+
+    solution = problem.solve(solver="cvxopt")
```

Comment:
As suggested with the `verbosity` argument, I wonder if we want to pass in the solver as a default argument to the function with a default value of `"cvxopt"`.